src/Form/ButtonsType.php

```diff
@@ -0,0 +1,59 @@
+<?php declare(strict_types=1);
+/**
+ * @copyright Zicht Online <https://zicht.nl>
+ */
+
+namespace Zicht\Bundle\AdminBundle\Form;
+
+use Symfony\Component\Form\AbstractType;
+use Symfony\Component\Form\FormInterface;
+use Symfony\Component\Form\FormView;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+class ButtonsType extends AbstractType
+{
+    /**
+     * Option 'buttons' should be an array of buttons. Each button can have the options as show below.
+     * Add a field with `'mapped' => false` to not map the buttons to any field of your subject.
+     * @example:
+     *   ->add(
+     *       '_action',
+     *       ButtonsType::class,
+     *       [
+     *           'mapped' => false,
+     *           'required' => false,
+     *           'buttons' => [
+     *               'key' => [
+     *                  // Button appearance options, all optional. Will fall back to default values when omitted.
+     *                   'label' => 'action_edit', # Optional. If not specified, the 'key' will be used as label
+     *                   'label_translation_domain' => 'SonataAdminBundle', # Optional. Will take translation domain of admin when left out
+     *                   'style' => 'danger', # Optional. Sonata button style (e.g. 'default' [= default], 'primary', 'success', 'info', 'danger', 'warning')
+     *                   'size' => 'sm', # Optional. Sonata button size. Skip for default size, otherwise pick 'xs', 'sm' or 'lg'.
+     *                   'icon' => 'pencil', # Optional. Sonata FA icon style
+     *                   'disabled' => 'true', # Optional. Set 'disabled' attribute on which Sonata applies a disabled style
+     *                  // Button action/target options. One of either of these is required
+     *                   'route' => 'edit', # Admin route to use to render the button URL, OR...
+     *                   'url' => '/admin/dashboard', # ... Abstract or relative URL to use
+     *               ],
+     *           ],
+     *       ]
+     *   )
+     */
+    public function configureOptions(OptionsResolver $resolver): void
+    {
+        $resolver->setRequired(['buttons']);
+        $resolver->setAllowedTypes('buttons', 'array[]');
+    }
+
+    public function buildView(FormView $view, FormInterface $form, array $options): void
+    {
+        $view->vars['buttons'] = $options['buttons'];
+
+        $view->vars['admin'] = $view->vars['sonata_admin']['admin'];
+    }
+
+    public function getBlockPrefix(): string
+    {
+        return 'zicht_' . parent::getBlockPrefix();
+    }
```

Comment:
Het is wel gebruikelijk dat de block prefix de naam van de classe volgt, dus lijkt me dan handiger om de class dan te hernoemen, dan staat de block prefix automatisch goed, i.e. `class ZichtButtonsType`.  Misschien zelfs nog het woordje Admin erbij?

Reply:
De classname zit al in een Zicht Admin(Bundle) namespace. Dus die behoeft geen extra toevoegingen. Ik zou ook niet een classname gaan aanpassen zodat ik de juiste block prefix krijg.

Ik zat inderdaad wel te twijfelen of er ook niet het woordje/prefix `admin_` bij moest in de block name. Want hij is wel echt alleen voor de admin geschikt.

Wat ik overal bij Zicht form types zie is dat de block prefix er hardcoded in staat, terwijl deze wordt afgeleid van de classname. Dus ik vond dit wel een mooie oplossing.

Reply:
Aangepast naar een hardcoded `'zicht_admin_buttons'`